backend/services/location.service.test.js

```diff
@@ -0,0 +1,78 @@
+const locationService = require('./location.service');
+const { setupDB } = require('../setup-test');
+const { Location } = require('../models/dictionaries/location.model');
+
+setupDB('location-service');
+
+describe('Test LocationService.add', () => {
+  test('Add empty array should throw database error', async (done) => {
+
+
+    try {
+      // Add empty array
+      await locationService.add([]);
+
+    } catch (err) {
+      expect(err.name).toBe('DatabaseError');
+    }
+
+    done();
+  });
+  test('Add an array of location strings sucessfully adds to database', async (done) => {
+    const locationsArray = ['loc1', 'loc2', 'loc3'];
+    await locationService.add(locationsArray);
+    const savedLocations = await locationService.getAll();
+
+    expect(savedLocations.length).toEqual(3);
+    expect(savedLocations).toEqual(expect.arrayContaining(locationsArray));
+
+    done();
+  });
+  test('Add duplicate string to location array should skip silently', async (done) => {
+    
+    const isArrayUnique = arr => Array.isArray(arr) && new Set(arr).size === arr.length;
+    await locationService.add(['loc1', 'loc2', 'loc3']);
+    const locationsArray = ['loc3', 'loc4', 'loc5'];
+    await locationService.add(locationsArray);
+    const savedLocations = await locationService.getAll();
+
+    expect(isArrayUnique(savedLocations)).toBeTruthy();
+    expect(savedLocations.length).toEqual(5);
+    expect(savedLocations).toEqual(expect.arrayContaining(locationsArray));
```

Comment:
please check out that result contains locations 1-5

Reply:
result contains 1-5